src/traces/bar/plot.js

```diff
@@ -58,8 +58,8 @@ function getXY(di, xa, ya, isHorizontal) {
     return isHorizontal ? [s, p] : [p, s];
 }
 
-function transition(selection, opts, makeOnCompleteCallback) {
-    if(hasTransition(opts)) {
+function transition(selection, fullLayout, opts, makeOnCompleteCallback) {
+    if(!fullLayout.uniformtext.mode && hasTransition(opts)) {
```

Comment:
Why? I thought you said this transitions looked fine

Reply:
They did not apply a uniform text size during and after smooth transition.